packages/jellyfish-api-core/__tests__/category/loan/createLoanScheme.test.ts

```diff
@@ -0,0 +1,112 @@
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nminimum collateral ratio cannot be less than 100\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if interestRate is less than 0.01', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(0.00999), { id: 'scheme2' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\ninterest rate cannot be less than 0.01\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if same minColRatio and interestRate were created before', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(100, new BigNumber(1.5), { id: 'scheme2' }) // NOTE(jingyi2811): Failed because its minColRatio and interestRate are same as default
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme default with same interestrate and mincolratio already exists\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if same id was created before', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: 'default' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme already exist with id default\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if id is an empty string', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: '' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: createloanscheme')
+  })
+
+  it('should not createLoanScheme if id is more than 8 chars long', async () => {
+    const promise = testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: '123456789' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: createloanscheme')
+  })
+
+  it('should createLoanScheme with utxos', async () => {
+    const masternodes = await testing.rpc.masternode.listMasternodes()
+    let masternodeId = ''
+    for (const id in masternodes) {
+      const masternode = masternodes[id]
+      if (masternode.mintedBlocks > 0) { // NOTE(jingyi2811): Find masternode that mined at least one block
+        masternodeId = id
+        break
+      }
+    }
+
+    const utxos = await container.call('listunspent', [1, 9999999, [masternodes[masternodeId].ownerAuthAddress]])
+    const inputs: UTXO[] = utxos.map((utxo: UTXO) => {
+      return {
+        txid: utxo.txid,
+        vout: utxo.vout
+      }
+    })
+
+    const loanSchemeId = await testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: 'scheme1', utxos: inputs })
```

Comment:
Improve the code, because the utxos is always an empty array.

```
const address = await container.getNewAddress()
const utxos = await container.call('listunspent', [1, 9999999, [address], true])
```

As the snippet above is copied from Oracle rpcs. Will fix them one day

Reply:
```
const utxo = await testing.container.fundAddress(await container.call('getnewaddress'), 10)
const loanSchemeId = await testing.rpc.loan.createLoanScheme(200, new BigNumber(2.5), { id: 'scheme1', utxos: [utxo] })
```
is an option

Reply:
You think what I thought before. fundAddress will raise `not from foundation member` error

Reply:
Ok I see, add a few comments to describre that it's using utxos from foundation member

Reply:
or better:
```
const utxo = await testing.container.fundAddress(GenesisKeys[0].owner.address, 10) // Funds foundation address
```

Reply:
Good suggestion